internal/v1/handler_blueprints.go

```diff
@@ -73,6 +105,32 @@ func (h *Handlers) CreateBlueprint(ctx echo.Context) error {
 	})
 }
 
+func (h *Handlers) UpdateBlueprint(ctx echo.Context) error {
+	var blueprintRequest CreateBlueprintRequest
+	err := ctx.Bind(&blueprintRequest)
+	if err != nil {
+		return err
+	}
+
+	blueprint := BlueprintFromAPI(blueprintRequest)
+	body, err := json.Marshal(blueprint)
+	if err != nil {
+		return err
+	}
+
+	id := uuid.New()
+	versionId := uuid.New()
+	err = h.server.db.UpdateBlueprint(versionId, id, body)
+	if err != nil {
+		logrus.Error("Error updating blueprint in db", err)
```

Comment:
What's the difference between using logrus and ctx.Logger for logging things? And should some of these other error paths have logging?

Reply:
To be honest, I do not have an idea, I just saw this in the repository few times. Is there a convention on when do you log things in certain way?

Reply:
The `ctx` variable is `echo.Context` type, that is a web request context from the Echo framework. It has its own logging facade. Ultimately, it is configured to use `logrus` logging library which also can be used directly, however, there is no context handler when used this way.

Therefore, if Echo context is available one should use `ctx.Logger()` which carries over request id and org id as well as other fields which is important for correlation in the logging database. This should be preferred and the context should be passed into as many functions as possible. It also supports conversion to standard library `context.Context` type which can be passed into other libraries like `pgx` (Postgres driver) to take full advantage of other context features. However, logging in functions which do not have `ctx` as an argument must be done directly, thus, these messages will not have correlation information.